rbac/ledger_sync/inbound/listener.py

```diff
@@ -0,0 +1,128 @@
+                            r.expr(
+                                {
+                                    "address": rec["address"],
+                                    "object_type": rec["object_type"],
+                                    "object_id": rec["object_id"],
+                                    "metadata": rec["metadata"],
+                                    "created_at": rec["timestamp"],
+                                    "updated_at": rec["timestamp"],
+                                }
+                            ),
+                            doc.merge(
+                                {
+                                    "metadata": rec["metadata"],
+                                    "updated_at": rec["timestamp"],
+                                }
+                            ),
+                        )
+                    )
+                )
+                result = database.run_query(query)
+                if (not result["inserted"] and not result["replaced"]) or result[
+                    "errors"
+                ] > 0:
+                    LOGGER.warning(
+                        "error updating metadata record:\n%s\n%s", result, query
+                    )
+            database.run_query(database.get_table("inbound_queue_archive").insert(rec))
+            database.run_query(
+                database.get_table("inbound_queue").get(rec["id"]).delete()
+            )
+        else:
+            rec["error"] = status[0]["invalid_transactions"][0]["message"]
+            database.run_query(database.get_table("inbound_queue_errors").insert(rec))
+            database.run_query(
+                database.get_table("inbound_queue").get(rec["id"]).delete()
+            )
+    except Exception as err:  # pylint: disable=broad-except
+        LOGGER.exception(
+            "%s exeption proccessing inbound record:\n%s", type(err).__name__, rec
+        )
+        LOGGER.exception(err)
+
+
+def listener():
+    """ Listener for Sawtooth State changes
+    """
+    try:
+        database = Database()
+        database.connect()
+
+        LOGGER.info("Reading queued Sawtooth transactions")
+        while True:
+            feed = database.run_query(database.get_table("inbound_queue"))
+            count = 0
+            for rec in feed:
+                process(rec, database)
+                count = count + 1
+            if count == 0:
+                break
+            LOGGER.info("Processed %s records in the inbound queue", count)
+        LOGGER.info("Listening for incoming Sawtooth transactions")
+        feed = database.run_query(database.get_table("inbound_queue").changes())
+        for rec in feed:
+            if rec["new_val"] and not rec["old_val"]:  # only insertions
+                process(rec["new_val"], database)
+
+    except Exception as err:  # pylint: disable=broad-except
+        LOGGER.exception("Outbound listener %s exception", type(err).__name__)
```

Comment:
This should be Inbound listener %s exception

Reply:
resolved